pages/_app.js

```diff
@@ -16,15 +16,19 @@ export default function App({ Component, pageProps }) {
     (initialEntry) => initialEntry.type === "InnerSituationAnalysis"
   );
 
-  function handleDelete(id) {
+  function handleDelete(id, currentPage) {
     confirmAlert({
       message: "Sicher, dass du diesen Eintrag löschen willst?",
       buttons: [
         { label: "Abbrechen" },
         {
           label: "Löschen",
-          onClick: () =>
-            setAllEntries(allEntries.filter((entry) => entry.id !== id)),
+          onClick: () => {
+            setAllEntries(allEntries.filter((entry) => entry.id !== id));
+            if (currentPage === "entry") {
+              history.back();
+            }
+          },
```

Comment:
you are using this function in one place and hardcorded "entry" for `currentPage` then you are asking for it: `if (currentPage === "entry")` so this will always be true.

I am not sure what history.back() does and what currentPage is used for, can you explain what is happening there, so I can better understand your reasoning?

Reply:
yes but I am using the handle delete function also on pages where currentpage is not "entry". this happens only in my entry-component. 
so when i delete from the entry page itself i want the browser to go one page back, so it doesn't stay on the deleted entry. 
Later I want to implement, that it looks what the last page was and to go there, if it belongs to the app. if it is another page (like e.g. google.com), than it should redirekt to the home-screen.

the handleDelete function is also used on my list-entry pages. there I don't need the "history.back()" function.